packages/hash/api/src/graph/system-org.ts

```diff
@@ -0,0 +1,123 @@
+export const ensureSystemOrgAccountIdExists = async (params: {
+  graphApi: GraphApi;
+  logger: Logger;
+}) => {
+  const { graphApi, logger } = params;
+  const { data: existingOrgEntitiesSubgraph } =
+    await graphApi.getEntitiesByQuery({
+      filter: {
+        all: [
+          { equal: [{ path: ["version"] }, { parameter: "latest" }] },
+          {
+            equal: [
+              { path: ["type", "versionedUri"] },
+              { parameter: types.entityType.org.entityTypeId },
+            ],
+          },
+        ],
+      },
+      graphResolveDepths: {
+        inheritsFrom: { outgoing: 0 },
+        constrainsValuesOn: { outgoing: 0 },
+        constrainsPropertiesOn: { outgoing: 0 },
+        constrainsLinksOn: { outgoing: 0 },
+        constrainsLinkDestinationsOn: { outgoing: 0 },
+        isOfType: { outgoing: 0 },
+        hasLeftEntity: { outgoing: 0, incoming: 0 },
+        hasRightEntity: { outgoing: 0, incoming: 0 },
+      },
+    });
+
+  const existingOrgEntities = getEntities(
+    existingOrgEntitiesSubgraph as Subgraph<SubgraphRootTypes["entity"]>,
+  );
+
+  const existingSystemOrgEntity = existingOrgEntities.find(
+    ({ properties }) =>
+      properties[
+        extractBaseUri(types.propertyType.shortName.propertyTypeId)
+      ] === SYSTEM_ACCOUNT_SHORTNAME,
+  );
+
+  if (existingSystemOrgEntity) {
+    systemOrgAccountId = extractEntityUuidFromEntityId(
+      existingSystemOrgEntity.metadata.editionId.baseId,
+    );
+    logger.info(`Using existing system org account id: ${systemOrgAccountId}`);
+  } else {
+    systemOrgAccountId = (await graphApi.createAccountId()).data;
+    logger.info(`Created system org account id: ${systemOrgAccountId}`);
+  }
+};
+
+// eslint-disable-next-line import/no-mutable-exports
+export let systemOrgModel: OrgModel;
+
+/**
+ * Ensure the `systemOrgModel` exists by fetching it or creating it using
+ * the `systemOrgAccountId`. Note this method must be run after the
+ * `systemOrgAccountId` and the system types have been initialized.
+ */
+export const ensureSystemOrgExists = async (params: {
+  graphApi: GraphApi;
+  logger: Logger;
+}) => {
+  const { graphApi, logger } = params;
+
+  const existingSystemOrgModel = await OrgModel.getOrgByShortname(graphApi, {
+    shortname: SYSTEM_ACCOUNT_SHORTNAME,
+  });
+
+  if (existingSystemOrgModel) {
+    systemOrgModel = existingSystemOrgModel;
+  } else {
+    systemOrgModel = await OrgModel.createOrg(graphApi, {
+      name: SYSTEM_ACCOUNT_NAME,
+      shortname: SYSTEM_ACCOUNT_SHORTNAME,
+      providedInfo: {
+        orgSize: OrgSize.ElevenToFifty,
+      },
+      actorId: systemOrgAccountId,
+    });
+
+    logger.info(
+      `System Org available with shortname = "${systemOrgModel.getShortname()}"`,
+    );
+
+    const pageTitles: PageDefinition[] = [
+      { title: "First" },
+      { title: "Second" },
+      { title: "Third" },
+    ];
+
+    await seedPages(pageTitles, systemOrgModel.getEntityUuid(), params);
```

Comment:
Should we be doing this here? I'd like to see a clear separation of seeding the required thing for the app to run (the org), and random test data (seeded pages)

Reply:
This was addressed as part of introducing the system user:
- https://github.com/hashintel/hash/pull/1596/commits/2101b462063712aa6e9ab795f2db22ec59236864
- https://github.com/hashintel/hash/pull/1596/commits/7634f6395ff669efabfaed4fc34c28a02aceea86